fuzzamoto-cli/src/commands/ir.rs

```diff
@@ -168,6 +185,80 @@ pub fn generate_ir(
     Ok(())
 }
 
+fn default_generators(context: &FullProgramContext) -> Vec<Box<dyn Generator<ThreadRng>>> {
+    vec![
+        Box::new(AdvanceTimeGenerator::default()),
+        Box::new(HeaderGenerator::new(context.headers.clone())),
+        Box::new(AddrRelayGenerator::new(context.addresses.clone())),
+        Box::new(AddrRelayV2Generator::new(context.addresses.clone())),
+        Box::new(GetAddrGenerator::default()),
+        Box::new(BlockGenerator::default()),
+    ]
+}
+
+fn instantiate_generators(
+    names: &[String],
+    context: &FullProgramContext,
+) -> Result<Vec<Box<dyn Generator<ThreadRng>>>> {
+    let factories = generator_factories();
+    let mut selected = Vec::new();
+    for name in names {
+        if let Some(factory) = factories
+            .iter()
+            .find(|spec| spec.name.eq_ignore_ascii_case(name))
+        {
+            selected.push((factory.builder)(context));
+        } else {
+            let available = factories
+                .iter()
+                .map(|spec| spec.name)
+                .collect::<Vec<_>>()
+                .join(", ");
+            return Err(CliError::InvalidInput(format!(
+                "Unknown generator '{name}'. Available generators: {available}"
+            )));
+        }
+    }
+    Ok(selected)
+}
+
+fn generator_factories() -> Vec<GeneratorFactory> {
```

Comment:
Do we really need the factories? Could we instead use the default generator list?